### ISSUE TITLE
Getting a panic when using the `--root` option

### ISSUE THREAD

**Issue by valmirjunior0088** [flamegraph-rs/flamegraph#99]
Hello!

I've been trying to run `cargo flamegraph`.

Unprivileged, it gave me an error complaining about `kernel.perf_event_paranoid = -1`, which I promptly solved by creating a `/etc/sysctl.d/10-cargo-flamegraph.conf` file with the aforementioned line.

After that, it began giving me the following warning:

```
WARNING: Kernel address maps (/proc/{kallsyms,modules}) are restricted,
check /proc/sys/kernel/kptr_restrict and /proc/sys/kernel/perf_event_paranoid.

Samples in kernel functions may not be resolved if a suitable vmlinux
file is not found in the buildid cache or in the vmlinux path.

Samples in kernel modules won't be resolved at all.

If some relocation was applied (e.g. kexec) symbols may be misresolved
even with a suitable vmlinux or kallsyms file.

Couldn't record kernel reference relocation symbol
Symbol resolution may be skewed if relocation was used (e.g. kexec).
Check /proc/kallsyms permission or run as root.
```

A flamegraph was generated but there was mostly just `[unknown]` sprinkled all over the graph.

At this point, I ran into the `--root` option, which seemed like a good option to bypass all these problems. But when I used it, another entirely different problem happened:

```
[ perf record: Woken up 1 times to write data ]
[ perf record: Captured and wrote 0.180 MB perf.data (20 samples) ]
writing flamegraph to "flamegraph.svg"
thread 'main' panicked at 'unable to generate a flamegraph from the collapsed stack data: Io(Custom { kind: InvalidData, error: "No stack counts found" })', /home/valmirpretto/.cargo/registry/src/github.com-1ecc6299db9ec823/flamegraph-0.3.1/src/lib.rs:273:6
```

Is there something I'm missing?

I'm trying to run `cargo flamegraph` on an Ubuntu machine, if that's relevant.

**Comment by szarykott** [comment]
I am having exactly the same problem on Linux Mint. Any workaround or solution?

**Comment by germolinal** [comment]
Same in Elementary OS

**Comment by djc** [comment]
I'm happy to mentor someone who wants to figure out a fix for this; unfortunately I don't currently have time to work on this myself unless there's some funding attached to it.

**Comment by germolinal** [comment]
Very understandable! I would love to help, but I don't have any sort of expertise to try to fix this.

Best!

**Comment by jscarrott** [comment]
As a workaround for this, I ended up using [inferno](https://github.com/jonhoo/inferno) directly following the instructions in its readme.

It's caused by perf.data ending up being owned by root with the current user not having any permissions to read the file. I've made a fix locally to chmod the file if you are using the root option but it is very hacky, I'll make a PR for it if I get some time to tidy it up.
